src/wallet/wallet.h

```diff
@@ -274,6 +274,15 @@ class CReserveKey
     //! Destructor. If a key has been reserved and not KeepKey'ed, it will be returned to the keypool
     ~CReserveKey()
     {
+#ifdef DEBUG
+        // To avoid accidental key re-use, we should attempt to have the creator
+        // have it returned or kept before its destructor is called.
+        // TODO: Fix the lifecycle of this object to be less error-prone!
+        assert(nIndex == -1);
+#endif
+        if (nIndex != -1) {
+            LogPrintf("Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible, e.g. here: " PACKAGE_BUGREPORT "/new\?title=wallet:%%20ReserveKey%%20was%%20not%%20dealt%%20with%%20until%%20destructor\n");
```

Comment:
Sorry for continued nit-picking, but the logprint could be made even if we assert out in the destructor.

```suggestion
            LogPrintf("Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible, e.g. here: %s%s\n", PACKAGE_BUGREPORT, "/new\?title=wallet:%20ReserveKey%20was%20not%20dealt%20with%20until%20destructor");
#ifdef DEBUG
            assert(false);
#endif
```

Reply:
indeed, a run of RPC tests appears to fail and it'd be nice to have it be obvious why